app/code/Magento/Ui/Component/Filters/Type/Input.php

```diff
@@ -70,7 +70,7 @@ protected function applyFilter()
             if (!empty($value)) {
                 $filter = $this->filterBuilder->setConditionType('like')
                     ->setField($this->getName())
-                    ->setValue(sprintf('%%%s%%', $value))
+                    ->setValue(sprintf('%%%s%%', str_replace('\\', '\\\\', $value)))
```

Comment:
I don't think it's responsibility of `Magento_Ui` to be aware of LIKE queries behavior, like necessity of double backslashes. Such replacement should occur as close to database layer as possible.

For example, in  `create` method of `\Magento\Framework\Api\FilterBuilder`, or even `\Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor::addFilterGroupToCollection` / `\Magento\Framework\Data\Collection\AbstractDb::addFieldToFilter`.

Also, please implement integration test enforcing this case.

Reply:
Agreed, that was easy but not proper place to put the logic. I didn't find a good place to put it in the mentioned classes so it applied just to the LIKE case and moved it to adapter. Let me know if that's ok

What would be the best place to create an integration test? Are there filters tests already somewhere?